src/transforms/code-scanning-alert.ts

```diff
@@ -35,50 +49,54 @@ const transformStatusToAppearance = (status: string): JiraRemoteLinkStatusAppear
 		case "dismissed":
 			return "moved"; // yellow
 		default:
+			logger.info(`Received unknown status from code_scanning_alert webhook: ${status}`);
 			return "default";
 	}
 }
 
-export default (context: Context): JiraRemoteLinkData => {
-	const { action, alert, ref, repository} = context.payload;
+export default (context: Context): Promise<JiraRemoteLinkData> => {
+	const {action, alert, ref, repository} = context.payload;
 
 	// Grab branch names or PR titles
-	const entityTitles = [];
+	const entityTitlesPromises: Promise<string>[] = [];
 	if (action === "closed_by_user" || action === "reopened_by_user") {
 		// These are manual operations done by users and are not associated to a specific Issue.
 		// The webhook contains ALL instances of this alert, so we need to grab the ref from each instance.
-		entityTitles.push(...alert.instances.map((instance) => getEntityTitle(instance.ref)));
+		entityTitlesPromises.push(...alert.instances.map((instance) => getEntityTitle(instance.ref, repository.name, repository.owner.login, context)));
 	} else {
 		// The action is associated with a single branch/PR
-		entityTitles.push(getEntityTitle(ref));
+		entityTitlesPromises.push(getEntityTitle(ref, repository.name, repository.owner.login, context));
 	}
 
-	const parser = issueKeyParser();
-	// Flatten and get all issue keys
-	const issueKeys = [].concat(...entityTitles.map((entityTitle) => parser.parse(entityTitle)));
+	return Promise.all(entityTitlesPromises).then(entityTitles => {
+		const parser = issueKeyParser();
+		const issueKeys = [].concat(...entityTitles.map((entityTitle) => parser.parse(entityTitle)));
+		if (issueKeys.length === 0) {
+			return undefined;
+		}
 
-	if (issueKeys.length === 0) {
-		return undefined;
-	}
-
-	return {
-		remoteLinks: [{
-			schemaVersion: "1.0",
-			id: `${repository.id.toString()}-${alert.number}`,
-			updateSequenceNumber: Date.now(),
-			displayName: `Alert #${alert.number}`,
-			description: alert.rule.description.substring(0, MAX_STRING_LENGTH),
-			url: alert.html_url,
-			type: "security",
-			status: {
-				appearance: transformStatusToAppearance(alert.most_recent_instance.state),
-				label: alert.most_recent_instance.state
-			},
-			lastUpdated: alert.created_at,
-			associations: [{
-				associationType: "issueKeys",
-				values: issueKeys
+		return {
+			remoteLinks: [{
+				schemaVersion: "1.0",
+				id: `${repository.id.toString()}-${alert.number}`,
+				updateSequenceNumber: Date.now(),
+				displayName: `Alert #${alert.number}`,
+				description: alert.rule.description.substring(0, MAX_STRING_LENGTH),
+				url: alert.html_url,
+				type: "security",
+				status: {
+					appearance: transformStatusToAppearance(alert.most_recent_instance.state),
+					label: alert.most_recent_instance.state
```

Comment:
Weren't we going to transform this too? I.e. `closed_by_user` to `dismissed` etc?

Reply:
Wait nvm, I got mixed up with action name vs state